.github/workflows/leo-build-tag-publish-and-run-tests.yml

```diff
@@ -0,0 +1,191 @@
+# Github action to Build Leonardo, add/promote semantic tagging and then run tests
+
+name: leo-build-tag-publish-and-run-tests
+
+on:
+  pull_request:
+    branches: [ develop ]
+    paths-ignore: [ '**.md' ]
+  push:
+    branches: [ develop ]
+    paths-ignore: [ '**.md' ]
+
+jobs:
+  leo-build-tag-publish-job:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    outputs:
+      custom-version-json: ${{ steps.render-leo-version.outputs.custom-version-json }}
+      tag: ${{ steps.tag.outputs.tag }}
+    steps:
+      - uses: 'actions/checkout@v4'
+
+      - name: Bump the tag to a new version
```

Comment:
Maybe a naive question, but we will now have two sets of GHA, one for GCP and one for Azure that would be tagged the exact same way, would this cause issues?

Reply:
*looks around*
Oh, hey I didn't realize the azure test were running their own builds. Thanks for the heads up! My thoughts here is that this will build on dev merge so it'll always be up to date, I can update the azure tests to use the version on dev, which should be the latest available version.